packages/@eventual/core-runtime/src/transaction-executor.ts

```diff
@@ -179,10 +174,46 @@ export function createTransactionExecutor(
                   normalizedKey
                 );
 
+                /**
+                 * When a set or delete is performed with an explicit expected version, immediately validate that the
+                 * entity resolved has that expected version.
+                 *
+                 * If a set or delete has already been performed, we'll replace that operation with this one
+                 * so we always use the original version.
+                 */
+                const expectedVersion = eventual.params[1]?.expectedVersion;
+                if (
+                  eventual.params[1]?.expectedVersion &&
```

Comment:
You're getting `eventual.params[1]?.expectedVersion` twice, use `expectedVersion` instead